\documentclass{article}

\usepackage[a4paper, total={6in, 8in}]{geometry}
\usepackage{setspace}
\usepackage{lineno}
\usepackage[dvipsnames]{xcolor}
\usepackage{graphicx}
\usepackage[square,sort,comma,numbers]{natbib}
\setcitestyle{authoryear,open={(},close={)}}

\title{Understanding the effects of date rounding in phylodynamics}
\author{Leo A. Featherstone$^{\ast,1}$, [Order TBA], Sebastian Duchene$^{\dagger,1}$}

\begin{document}

\maketitle
\linenumbers
$^{1}$ Peter Doherty Institute for Infection and Immunity, University of Melbourne, Australia.\\
*email: leo.featherstone@unimelb.edu.au

\section*{Abstract}
\textbf{Do at end, need to be Public-Healthy-y}

\begin{spacing}{1.5}
\section*{Introduction}
\textcolor{red}{Sharing pathogen sequence data ...}
Pathogen genomics has played an increasingly important role in our understanding of infectious outbreaks, including major pandemics, such as those of SARS-CoV-2, Ebola virus, \textit{Mycobacterium tuberculosis} (the bacterium responsible for tuberculosis, abbreviated as TB), and drug resistant bacteria \citep{lancet2021genomic}. Phylodynamic tools that have seen a surge in adoption, particularly since the 2013-2016 West African Ebola outbreak. This outbreak was the first instance in which genome sequence data were generated as the outbreak unfolded, and thus phylodynamic inferences could be used to inform public health responses \citep{mbala2019medical}.

Phylodynamic methods can draw a range of inferences from pathogen sequence data, including transmission parameters, the direction and rate of geographic movement, and the time and location of origin of infectious outbreaks \citep{featherstone2022epidemiological, attwood2022phylogenetic, du2015getting}. The basis of phylodynamic analyses is that epidemiological spread leaves a trace in the genomes of pathogen populations, in the form of substitutions or other molecular information. Such pathogen population are also known as `measurably evolving populations' \citep{drummond2003measurably, biek_measurably_2015}.

Phylodynamic models that estimate epidemiological parameters such as the effective reproductive number $R_e$ exploit substitutions and sequence sampling times \citep{featherstone_decoding_2023}. As a consequence, pathogens for which the timescale of transmission coincides with the timescale over which they acquire substitutions are particularly well suited for phylodynamic analyses. As a case in point, H1N1 influenza virus accumulates substitutions at a rate of about 4 $\times10^{-3}$ subs/site/year \citep{hedge_2013_real-time}. Because its genome length is around 13,158 bp, we would expect 0.06 substitutions over the course of an infection ($\sim$ 4 days) and one substitution to appear every 90 days. Clearly, providing sampling times with a precision of a year would remove valuable information about the molecular and epidemiological dynamics, which occur over a timescale shorter than the unit of rounding and thus potentially introduce bias to phylodynamic estimates. In contrast, the evolutionary rate for TB has been estimated to be of the order of $10^{-8}$ subs/site/year \citep{menardo2019molecular}, resulting in about 0.043 substitutions per year (for a genome length of 4.3 Mbp), one substitution every 23 years, and an expected 0.34 substitutions over the course of an infection (8 years) \citep{kuhnert_tuberculosis_2018}. For this bacterium, providing the sampling year or month may be sufficiently precise to correctly inform phylodynamic analyses.

Ideally empirical sequence data sets should include precise sampling time information for all samples, with the day, month, and year of collection \citep{black2020ten}. Nevertheless sequence sampling dates are often considered part of the associated metadata and may be unavailable or imprecise for different reasons, including patient or organisation confidentiality or a lack of a consistent platform for storing these data \citep{raza2016big}. The amount of genome sequences for SARS-CoV-2 in the GISAID database is about 15.8M \citep{shu2017gisaid}, with about 2.4\% (382K) having `incomplete' date information, where no sampling time may be reported, or its precision may only include the year (verified early August 2023). 

Including these sequences with imprecise sampling times in phylodynamic analyses requires the researcher to assume that they have been sampled at an arbitrary day. Selecting the arbitrary day can be motivated by convenience, for instance with all samples from 2020 being assigned 1st January 2020 or 15 June 2020, or by sampling a random day within 2020 using a statistical distribution. In any case, this practice introduces a degree of error. Indeed,sequences sampled 11 months apart may be assigned the exact same day. 

\textcolor{blue}{Importantly, although phylodyanmic analyses benefit from using precise sampling times, they are agnostic to the actual calendar date. An exponentially growing population sampled at a constant rate in the year 1997 will have the same distribution of sequence sampling times and coalescent events as one from the year 2020. Thus, a straight-forward approach to encrypt sampling dates is to provide their the number of days between sampling times, and not the actual calendar dates. - Not sure if this should be mentioned here.}

Here, we investigated the impact of different degrees of precision in sampling times on phylodyanmic estimates of key parameters, including $R_e$, the molecular clock rate, and the time of origin of the outbreak. We considered a range of pathogens, H1N1 influenza, SARS-CoV-2, \textit{Shigella sonnei}, and TB. These organisms have undergone substantial genome surveillance and have different infectious periods and molecular evolutionary dynamics. To quantify the impact of date precision in phylodynamics, we conducted extensive simulations, where we are able to assess precision and accuracy in estimates of key parameters.


\textcolor{red}{Do we need short para on the phylodynamic threshold? It is kind of related but not quite the same. Maybe leave to the Discussion...}

\textcolor{red}{We we care about rounding? Here add about GISAID and other data bases and then the remaining items below. }


\begin{itemize}
    \item Increased sharing of pathogen genome sequences has been a feature of responses to recent infectious disease threats. This is also the culmination of a broader trend that has build with advances in WGS. 
    \item Examples of GISAID and other ID databases
    \item Patient confidentiality remains a key priority 
    \item Define date rounding practice, provide citations to the extensiveness of the practice (HELP!!) - get some sentences from Courtney
    \item Introduce how we tackle the problem
    \item Explain hypothesised and shown axis in the data between temporal clustering and inflated rates
    \item May introduce effective mutation time here?
    \item Mention that real-world data features blur the trends we expect, so mention that we conclude with the proposed encryption algorithm
\end{itemize}

\section*{Methods}
\subsection*{Overview}
Our study is based around 4 empirical datasets of H1N1 influenza virus, SARS-CoV-2, \textit{Shigella sonnei}, and \textit{Mycobacterium tuberculosis} and a corresponding simulation study. For both the empirical and simulated datasets, we performed phylodynamic analysis with sampling dates rounded to the day, month, year, and measure the resulting bias critical parameters - $R_0$ / $R_e$ and the age of the outbreak (origin hereafter). For example, two samples from 2000/05/14 and 2000/05/02 would become 2000/05/01, if rounded to the month. 

In this context $R_0$ refers to the \textit{basic} reproductive number and $R_e$ is the \textit{effective} reproductive number. These parameters correspond to the average number of secondary infections at the start of an outbreak (i.e. assuming a fully susceptible population; $R_0$) or thereafter ($R_e$) (reviewed by \citep{featherstone2022epidemiological, du2015getting, kuhnert2011phylogenetic}). We consider the origin parameter as the age of the root node and not the length of the root branch to facilitate comparison between models, in contrast to Stadler \citet{stadler2012estimating}, where the origin parameter includes the age of the root branch.

The two viral datasets consist of samples from the 2009 H1N1 pandemic (n=161) from \citet{hedge_2013_real-time}, and a cluster of early SARS-CoV-2 cases from  Australia in 2020 (n = 112) \citep{lane2021genomics}. The bacterial datasets consist of Australian \textit{S. sonnei} samples from an outbreak studied by \citet{ingle_co-circulation_2019}, and 36 \textit{M. tuberculosis} samples from a ~25 year outbreak studied by \citet{kuhnert_tuberculosis_2018}. These data were chosen because they encompass a diversity of epidemiological dynamics and scales with variable rates of substitution.

\subsection*{Simulation Study}
We simulated outbreaks as Birth-Death sampling processes using the Master package in BEAST v2.6.6 \citep{vaughan_stochastic_2013,bouckaert_beast_2019}. These simulations consisted of 100 replicates over 4 parameter sets the represent values for each of the empirical datasets. All parameter sets include a proportion of cases sequenced ($p$), duration ($T$), and a "becoming un-infectious" rate ($\delta$ = reciprocal of the duration of infection). For simulations corresponding the viral datasets, transmission is modelled via $R_0$, the average number of secondary infections. For those corresponding to the bacterial datasets, we allow the effective reproductive numbers to change after an interval of time, $R_{e_1}$ and $R_{e_2}$, with a change time at $0.5T$. This resulted in a total of 400 outbreak datasets which we then used to simulate sequence data under a Jukes-Cantor model using Seq-Gen v1.3.4 \citep{rambaut_seq-gen_1997}. Substitution rates, genome lengths, and the above outbreak parameters are summarised in tables \ref{tab:sim_parms} and \ref{tab:seq_parms}.

\begin{table}[ht]
    \centering
    \caption{Parameter sets outbreaks corresponding to each empircal dataset.}
    \begin{tabular}{l|c|c|c|c|c|c|l|}
    \hline
    Microbe                     &   $\delta (yrs)^{-1}$    & $R_0$ &   $R_{e_1}$   &  $R_{e_2}$    &   $p$   &   $T$(yrs)   & Source \\
    \hline
    H1N1                        &   91.31    & 1.3 &   -   &  -    &   0.015   &   0.25 & \citet{hedge_2013_real-time} \\
    SARS-CoV-2                  &   36.56    & 2.5 &   -   &  -   &   0.80   &  0.16 & \citet{lane2021genomics} \\
    \textit{Shigella sonnei}    &   52.18    &  - &   1.5   &  1.01   &   0.40   &   0.50 & \citet{ingle_co-circulation_2019} \\
    \textit{M. tuberculosis}    &   0.125    &  - &   2.0   &  1.10    &   0.08   &   25.0 & \citet{kuhnert_tuberculosis_2018} \\
    \hline
    \end{tabular}
    \label{tab:sim_parms}
\end{table}

\begin{table}[h!]
    \centering
    \caption{Substitution rates and genome length for sequence simulation.}
    \begin{tabular}{l|c|l|r}
    \hline
    Microbe                     &   Substitution Rate (subs/site/yr) & Genome Length & Time/Sub/Genome (yrs)  \\
    \hline
    H1N1                        & $4\times10^{-3}$ & 13158 & 0.0190\\
    SARS-CoV-2                  & $1\times10^{-3}$ & 29903 & 0.0334\\
    \textit{S. sonnei}    & $9\times10^{-7}$ & 4825265  & 0.3454\\
    \textit{M. tuberculosis}    &   $1\times10^{-7}$ & 4300000 & 23.256\\
    \hline
    \end{tabular}
    \label{tab:seq_parms}
\end{table}

\subsection*{Empirical Data}
We conducted Bayesian phylodynamic analyses were conducted using a Birth-Death skyline tree prior in BEAST v2.6.6 \citep{bouckaert_beast_2019}. We sampled from the posterior distribution using Markov chain Monte Carlo (MCMC), with length of $5\times10^{8}$ steps, with the initial 10\% discarded as burnin. To determine sufficient sampling from the stationary distribution we verified that the effective sample size (ESS) of key parameters was at least 200.

\subsubsection*{H1N1}
The H1N1 data consist of 161 samples from North America during the 2009 H1N1 influenza virus pandemic, analysed by \citet{hedge_2013_real-time}. This  dataset provides an example of a rapidly evolving pathogen sparsely sampled over a longer epidemiological timescale. 

We placed a $\textnormal{Lognormal}(\mu=0,\sigma=1)$ prior on $R_0$, $\beta(1,1)$ prior on $p$, and fixed the becoming-uninfectious ($\delta = 91$), corresponding to a 4 day duration of infection. We also placed an improper ($U(0,\infty)$) prior on the origin and a $U(10^{-4},10^{-2})$ prior on the substitution rate. This prior corresponds to analysis of these data in \citet{featherstone_decoding_2023}.

\subsubsection*{SARS-CoV-2}
The SARS-CoV-2 data are 112 samples from a densely sequenced transmission cluster in Victoria, Australia in 2020, first analysed by \citet{lane2021genomics}. These data are similar to the H1N1 datasets in presenting a quickly evolving viral pathogen, but contrast in that virtually all cases in the cluster were sequenced. 

Prior configurations are identical to those used in \citet{featherstone_decoding_2023} to analyse the same data. Briefly, we placed a 

$\textrm{Lognormal}(\textrm{mean}=1, \textrm{sd}=1.25)$ prior on $R_0$ and an $\textrm{Inv-Gamma}(\alpha=5.807, \beta=346.020)$ prior on the becoming-uninfectious rate ($\delta$).  The sampling proportion was fixed to $p=0.8$ since every known Victorian SARS-CoV-2 case was sequenced at this stage of the pandemic, with a roughly 20\% sequencing failure rate. We also placed an $\textrm{Exp}(\textrm{mean}=0.019)$ prior on the origin, corresponding to a lag of up to one week  between the index case and the first putative transmission event. The substitution rate was fixed a $10^{-3}$ following \citep{duchene_temporal_2020}.

\subsubsection*{\textit{Shigella sonnei}}
The \textit{S. Sonnei} dataset originates from \citet{ingle_co-circulation_2019} and consists of a single nucleotide polymorphism (SNP) alignment of 146 sequenced isolates from infected men who have sex with men in Australia. These data provide an example of densely sequenced bacterial outbreak. 

To accommodate changing transmission dynamics, we included two intervals for $R_e$ with a $\textnormal{Lognormal}(\mu=0,\sigma=1)$ prior on each. We also placed a $\beta(1,1)$ prior on the sampling proportion, a $U(0,1000)$ prior on the origin, and fixed the becoming un-infectious rate at $\delta=73.05$ corresponding to a 5 day duration of infection.

To generate the SNP alignment, we (Enter Danielle...)

\subsubsection*{\textit{Mycobacterium tuberculosis}}
The \textit{M. tuberculosis} dataset consists of 36 sequenced isolates taken from a retrospectively recognised outbreak in California, USA, and originating in the Wat Tham Krabok refugee camp in Thailand. We applied the same similar prior configuration to \citet{kuhnert_tuberculosis_2018}, with the exception of including 2 intervals for $R_e$ and fitting a strict molecular clock with a $\Gamma(\alpha=0.001,\beta=1000.0)$ prior.


\section*{Results}
\subsection*{Simulation study}
Broadly, the bias in posterior mean reproductive number increases with decreasing date resolution. This effect is most pronounced for the viral simulation conditions, where the rounding units of one month or one year is greater than the amount of time expected for one mutation to arise. In this case, date rounding condenses divergent sequences in time, driving a signal for higher rates of evolution and transmission. Conversely, the effect is less pronounced in the bacterial conditions where the date resolution lost is a smaller fraction of the effective mutation time, such as for the \textit{M. tuberculosis} and \textit{S.sonnei} data sets. In these cases, sequences are less divergent such that temporal clustering does not inflate posterior evolutionary rate. Moreover, the sampling timespans for these datasets are longer (table \ref{tab:sim_parms}), meaning that clustering to month or year leads to a less pronounced inflation of the reproductive number as samples still remain temporally distributed.

Corresponding with the above trend in evolutionary rate, the mean posterior origin time have an  upwards bias, representing a signal for a shorter outbreak duration (Fig \ref{fig:simOrigin}). This is the result of a well understood axis among phylodynamic models where higher rates of evolution suggest shorter periods of evolution \citep{featherstone_decoding_2023}. In the epidemiological view, this translates into placing more weight on lower values for the duration of the outbreak.

The H1N1 influenza virus simulation conditions demonstrate strongest relationship between high estimates of evolutionary rates and shorter outbreak duration. It can be thought of as the simulation conditions with the highest divergence among sequences relative to simulation time, owing to a combination of a higher mutation and transmission rate alongside a lower mutation rate (table \ref{tab:sim_parms}). For the date rounding to the year, we see extremely high values of $R_0$ and substitution, with means of around $10^{8}$ and $10^{6}$ subs/site/year, respectively. Such values, although implausible, demonstrate a key point that bias in posterior estimates compounds with decreasing date resolution. The effect is nonlinear, but also exacerbated by more divergent sequences, which would otherwise make for an idea phylodynamic dataset \citep{featherstone_decoding_2023}. Rounding to the month demonstrates intermediate effects with erroneously high bias. The SARS-Cov-2 simulation condition presents a similar trend, albeit with a less extreme degree of bias. 

The two bacterial simulation conditions demonstrate the same trends in $R_e$, the substitution rate, and the origin. The \textit{S. sonnei} dataset shows intermediate effects with minimal bias when moving to month resolution and larger effects at year levels for all the above parameters. This is expected given its effective mutation time is somewhere between the order of months and years (table \ref{tab:sim_parms}). This effect is also markedly increased for $R_{e_2}$ in comparison to $R_{e_1}$ at the year level, suggesting that bias also increases where more distinct samples appear to arise at the same time (we expect more samples in the second window of the \textit{S. sonnei} simulations).

The \textit{M. tuberculosis} simulation conditions effectively act as a control conditions, since it appears inter to date rounding. Again this is expected, because this dataset reflects both longer simulation time, with temporal clustering less likely to inflate $R_e$, but also the effective mutation time is longer than 1 year. As such, even rounding to a year is unlikely to drive a signal for increased evolutionary rate or a more recent origin time.

\begin{figure}
    \centering
    \includegraphics{sim_clock_trajectory.pdf}
    \caption{Mean posterior evolutionary rate for each simulation condition over decreasing date resolution. Lines connect individual simulated datasets across analyses with decreasing date resolution and horizontal black lines mark the true evolutionary rate. Mean posterior evolutionary rate increases where date rounding clusters more divergent sequences, such as in the case of the viral datasets. The effect is less pronounced for the slower evolving simulation conditions - (\textit{S. sonnei} and \textit{M. tuberculosis}).}
    \label{fig:simClock}
\end{figure}

\begin{figure}
    \centering
    \includegraphics{sim_Re_trajectory.pdf}
    \caption{Bias in $R_0$ or $R_e$ over decreasing date resolution for simulated data. Lines connect posterior mean reproductive number for individual simulated datasets analysed under decreasing date resolution under each simulation condition. Horizontal black lines show the true value. In general, the reproductive number biases upwares with decreasing date resolution, with the most dimished effects where the date resolution is a smaller fraction of average time required for a mutation (\textit{S. sonnei} and \textit{M. tuberculosis}).}
    \label{fig:simR0}
\end{figure}

\subsection*{Empirical Results}

\begin{table}[ht]
\centering
\begin{tabular}{rllrlrlrlrlrlrl}
  \hline
 & organism & resolution & meanR0 & R0HPD & meanRe1 & Re1HPD & meanRe2 & Re2HPD & meanP & pHPD & meanDelta & deltaHPD & meanOrigin & originHPD \\ 
  \hline
1 & H1N1 & Day & 1.083 & [1.05, 1.11] &  & [NA, NA] &  & [NA, NA] & 0.011 & [0.00656, 0.0158] &  & [NA, NA] & 0.417 & [0.344, 0.554] \\ 
  2 & H1N1 & Month & 1.144 & [1.11, 1.17] &  & [NA, NA] &  & [NA, NA] & 0.007 & [0.00362, 0.0104] &  & [NA, NA] & 0.419 & [0.338, 0.556] \\ 
  3 & H1N1 & Year & 1.154$\times 10^8$ & [$8.98\times10^7$, 1.45e+08] &  & [NA, NA] &  & [NA, NA] & 0.250 & [0.00203, 0.932] &  & [NA, NA] & 0.000 & [2.41e-09, 3.72e-09] \\ 
  4 & SARS-CoV-2 & Day & 1.207 & [0.919, 1.57] &  & [NA, NA] &  & [NA, NA] &  & [NA, NA] & 81.463 & [51.3, 122] & 0.150 & [0.143, 0.164] \\ 
  5 & SARS-CoV-2 & Month & 5.972 & [3.84, 9.21] &  & [NA, NA] &  & [NA, NA] &  & [NA, NA] & 97.499 & [62.3, 142] & 0.172 & [0.17, 0.176] \\ 
  6 & SARS-CoV-2 & Year & 18.689 & [10.5, 29.8] &  & [NA, NA] &  & [NA, NA] &  & [NA, NA] & 43.862 & [25.9, 70.8] & 0.143 & [0.142, 0.147] \\ 
  7 & Shigella & Day &  & [NA, NA] & 1.072 & [1.03, 1.11] & 0.982 & [0.968, 0.997] &  & [NA, NA] &  & [NA, NA] & 3.405 & [2.85, 3.62] \\ 
  8 & Shigella & Month &  & [NA, NA] & 1.073 & [1.03, 1.11] & 0.983 & [0.969, 0.997] &  & [NA, NA] &  & [NA, NA] & 3.408 & [3.14, 3.62] \\ 
  9 & Shigella & Year &  & [NA, NA] & 1.174 & [1.13, 1.22] & 0.949 & [0.933, 0.963] &  & [NA, NA] &  & [NA, NA] & 4.004 & [4, 4.02] \\ 
  10 & TB & Day &  & [NA, NA] & 2.492 & [0.688, 4.88] & 1.292 & [0.704, 2.4] & 0.089 & [0.043, 0.148] & 0.290 & [0.106, 0.614] & 22.984 & [16.3, 49.1] \\ 
  11 & TB & Month &  & [NA, NA] & 2.789 & [0.576, 5.15] & 1.390 & [0.735, 2.56] & 0.087 & [0.0417, 0.149] & 0.229 & [0.0993, 0.416] & 24.505 & [17.9, 52.1] \\ 
  12 & TB & Year &  & [NA, NA] & 2.751 & [0.5, 5.27] & 1.484 & [0.774, 2.84] & 0.086 & [0.04, 0.147] & 0.222 & [0.0949, 0.401] & 25.551 & [18, 55.2] \\ 
   \hline
\end{tabular}
\label{table:empData}
\end{table}

Broadly, analyses of the empirical datasets reproduce the trends of bias in reproductive number, substitution rate, and time of origin from the simulation study (figures \ref{fig:empR},\ref{fig:empClock}). That is, the reproductive number increases with decreasing date resolution along with an increase in the substitution rate and corresponding decrease in the origin. There are a few exceptions to this trend that we consider below and which we attribute to the difference between simulated and empirical sampling time distributions.

\subsubsection*{H1N1 influenza virus}
Posterior $R_0$ increases with decreasing date resolution in a comparable way to the simulation study. However, the posterior substitution rate and origin time estimates remain essentially the same for day and month resolution (mean values of 1.083, 1.44 and $10^{-2}$, $10^{-2}$ respectively)(table \ref{table:empData}),  before moving upwards at year resolution as expected from the simulation study. This finding can be explained by the sampling time distribution, since the earlier samples occur later in their month (change fig3 to date axis proper), such that rounding them down to the first of the month effectively expands the timespan of sampling and thus driving signal for a lower evolutionary rate and older origin time.

\subsubsection*{SARS-CoV-2}
The SARS-CoV-2 datasets behaves as expected with respect to the posterior $R_0$. In particular, rounding to the month results in an unlikely, but plausible value of $R_0 = 5.972$ (table \ref{table:empData}). Rounding to the year inflates $R_0$ further as expected.

The posterior substitution rate of SARS-CoV-2 remains essentially stable when rounding to the month or year, with a mean value of $10^{-3.5}$ (subs/site/time) for both (table \ref{table:empData}). In addition, the origin time at month resolution becomes older after rounding the sampling times. These findings stand in contrast to the expectation of rounding sampling times leading to an overestimate of the substitution rate and a corresponding underestimation of the time of origin. We again attribute these differences to the distribution of the empirical sampling times, which are not as consistently distributed as they are for the simulated outbreaks. There appears to be one early sample (Fig \ref{fig:empR} A) that likely drives the signal for an older outbreak when rounding to the month because it is pushed back in time. At the same time, the substitution rate increases with decreasing date resolution as expected, likely due to the clustering of the the rest of the samples after the earliest.

\subsubsection*{\textit{S. sonnei}}
For $R_{e_1}$, the \textit{S. sonnei} dataset matched the simulation study, with month rounding having a minimal effect, but year rounding inducing an upwards bias (mean values of 1.072, 1.073 respectively, figure \ref{fig:empR}). $R_{e_2}$ departs from expectation. The estimate for this parameter decreases when rounding to the year. We speculate that this occurs because it compensates for elevated $R_{e_1}$ earlier in the outbreak. This is supported by a markedly lower origin value (mean of 4.004, table \ref{table:empData}), such that the outbreak appears as an intensified early burst. The substitution rate remains stable across date resolutions, which is expected given the overall low substitution in this data set (around $10^{-6}$ subs/site/year).

\subsubsection*{\textit{M. tuberculosis}}
The \textit{M. tuberculosis} data recapitulate the outcome of the simulation study. Posterior origin times and evolutionary rate remain consistent across decreasing date resolution at 20 years and $10^{-7}$ (subs/site/time) respectively. We observe minimal upwards bias in posterior $R_{e_1}$ and $R_{e_2}$, and the expectation that $R_{e_1} > R_{e_s}$ is met, coinciding with an earlier burst of transmission in agreement with \citet{kuhnert_tuberculosis_2018}. This reaffirms that if the effective mutation time sufficiently large compared to the date resolution lost, then date rounding has a lesser effect.

\begin{figure}[h!]
    \centering
    \includegraphics{empirical_plot.pdf}
    \caption{Posterior reproductive number and origin for each empirical dataset coloured by level of date resolution. Posterior origin times are represented as rescaled posterior frequencies along the Date axis and posterior reproductive numbers are given in violin plots on the vertical axis. For the H1N1 and SARS-CoV-2 datasets, posterior $R_0$ across date resolution is overlayed and overlaps minimally. For the \textit{S. sonnei} and \textit{M. tuberculosis}, posterior $R_{e_1}$ and $R_{e_2}$ (left to right) are displayed in adjacent groups. The change time between them is itself variable as half of the origin time. Samplng times are given as black mark son each date axis.}
    \label{fig:empR}
\end{figure}

\begin{figure}[h!]
    \centering
    \includegraphics{empirical_clock_trajectory.pdf}
    \caption{Posterior substitution rate for each empirical dataset across analyses with decreasing date resolution.}
    \label{fig:empClock}
\end{figure}

\section*{Discussion}

The results of the simulation study can by summarised as showing that date rounding inflates estimates of evolutionary and epidemiological rates by temporally clustering differentiated genome sequences. This factor manifested as upwards bias of the effective reproductive number, substitution rate, and an underestimate of the age of the outbreak. The extent of the bias increased with more diverged sequences and decreased date resolution. In other words, it increases with the assertion that more evolution occurred in less time. This is why bias increased for simulation conditions with the highest amount of mutation per unit time - the H1N1 influenza virus conditions followed SARS-CoV-2 and \textit{S. sonnei}. Our \textit{M. tuberculosis} simulations, in their inertness to date rounding, also support this explanation because they were unlikely to generate any mutations over the month or even year mutation timescales.

Our empirical analyses broadly recapitulated the results of the simulation study, but also introduced notable exceptions which emphasised the unpredictability of the magnitude and direction of estimation bias when rounding dates. For example, the posterior substitution rate of the H1N1 influenza virus dataset did not display an upward bias when rounding to the month. The SARS-CoV evolutionary rate did not increase when moving from month to year rounding, and the posterior $R_{e_2}$ decreased when rounding to the year for the \textit{S. sonnei} dataset. In each case, we attribute these differences to the way in which the distribution of empirical sampling times differed from the consistency our our simulations. This meant that date rounding did not always result in temporal clustering of divergent sequences. In the example of the SARS-CoV-2 dataset where samples originated from the end of one month and start of the next, rounding down to the start of each month serves to spread out the samples over time, overriding the effect of clustering samples from the same month.

Taken together, the results form the simulation study and empirical data show that although date rounding biases empidemiological estimates in a theoretically predictable direction, the intensity of the bias is difficult to predict and varies with the parameter space the data notionally inhabit. Moreover, features of real-world sampling such as fine-scale clustering of sampling times over longer sampling efforts can unpredictably dampen or reverse expected bias due to date rounding. Put succinctly, date rounding induces bias unpredicable bias due to the interaction of theoretical aspects of phylodynamic models real-world data features.

We conclude that accurate sampling time information is essential where phylodynamic insight is needed to understand infectious disease epidemiology and evolution. There does not appear to be an clear way to adjust for the bias caused otherwise. However, as acknowledged from the beginning of this article, it may impose an unacceptable level of risk to patient confidentiality to release highly precise isolate sampling times, as can theoretically be used to identify individual patients. To circumvent this and deliver timely phylodynamic results, we finish by proposing an extremely simple form of encryption that may lower the level of risk in sharing sampling time to the day to acceptable levels.

\subsection*{The simplest encryption of dates}
The functional component of phylodynamic data is the \emph{difference} between sequences and dates, rather than their absolute values. After all, our methods are comparative within a sample. Thus we can prioritise exact information and protect patient identity at the same time. We propose that data deposited in online databases include dates that are all shifted in time by an unknown seed number, and reinterpret results by factoring this in. For example, if the sampling times of a dataset of 3 samples are 2000, 2001 and 2002, then we may randomly draw a seed of 1000 with which to shift and dates deposit online 2000, 2001 and 2002 $\rightarrow$ 3000, 3001 and 3002. Then results can be reinterpreted with regard to the random seed. If, for example the estimated time of onset was 3 years before the most recent sample, then those receiving the data will not be able to place this in time, while those on the data generation end can interpret this correctly (estimated time of onset = 2002-3 = 1999). In the same vein, transmission parameters such as $R_e$ can be understood to pertain to the true sampling time.

\end{spacing}

\bibliographystyle{natbib}
\bibliography{refs}

\subsection*{Supplementary Material}

\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}

\begin{figure}[h!]
    \centering
    \includegraphics{sim_origin_trajectory.pdf}
    \caption{Meas posterior origin for each simulation condition over decreasing date resolution. Lines connect individual simulated datasets across analyses with decreasing date resolution and horizontal black lines mark the true evolutionary rate. Mean posterior origin decreases  where date roundeding clusters more divergent sequences, such as in the case of the viral datasets. The effect is less pronounced for the slower evolving simulation conditions - (\textit{S. sonnei} and \textit{M. tuberculosis}).}
    \label{fig:simOrigin}
\end{figure}

\begin{figure}
    \centering
    \includegraphics{empirical_origin.pdf}
    \caption{Posterior origin time across date resolution and simulation conditions.}
    \label{fig:empOrigin}
\end{figure}

\end{document}